python/lsst/cp/pipe/cpFilterScan.py

```diff
@@ -0,0 +1,377 @@
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+import numpy as np
+import copy
+
+from astropy.table import Table
+
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+import lsst.pipe.base.connectionTypes as cT
+
+from . import CalibCombineTask, CpEfdClient
+from lsst.daf.base import PropertyList, DateTime
+
+__all__ = [
+    "CpFilterScanTask", "CpFilterScanTaskConfig",
+    "CpMonochromatorScanTask", "CpMonochromatorScanConfig"
+]
+
+
+class CpFilterScanConnections(pipeBase.PipelineTaskConnections,
+                              dimensions=("instrument", "detector")):
+    inputExpHandles = cT.Input(
+        name="postISRCCD",
+        doc="Input exposures to measure statistics from.",
+        storageClass="Exposure",
+        dimensions=("instrument", "physical_filter", "exposure", "detector"),
+        multiple=True,
+        deferLoad=True,
+    )
+    inputElectrometerHandles = cT.Input(
+        name="electrometer",
+        doc="Input electrometer data.",
+        storageClass="IsrCalib",
+        dimensions=("instrument", "exposure"),
+        multiple=True,
+        deferLoad=True,
+    )
+    outputData = cT.Output(
+        name="cpFilterScan",
+        doc="Output table to write.",
+        storageClass="ArrowAstropy",
+        dimensions=("instrument", "detector"),
+    )
+
+    def __init__(self, *, config=None):
+        super().__init__(config=config)
+
+        if config and not config.useElectrometer:
+            self.inputs.discard("inputElectrometerHandles")
+
+
+class CpFilterScanTaskConfig(pipeBase.PipelineTaskConfig,
+                             pipelineConnections=CpFilterScanConnections):
+    maskNameList = pexConfig.ListField(
+        dtype=str,
+        doc="Mask list to exclude from statistics calculations.",
+        default=["DETECTED", "BAD", "CR", "NO_DATA", "INTRP"]
+    )
+    useElectrometer = pexConfig.Field(
+        dtype=bool,
+        doc="Use electrometer data?",
+        default=False,
+    )
+    referenceFilter = pexConfig.Field(
+        dtype=str,
+        doc="Filter to use as baseline reference.",
+        default="empty~empty",
```

Comment:
I know this is the default right now for LATISS, but I think the dual naming is LATISS-only.  I would leave this as `empty` or better yet make it required to be overridden in the instrumental pipelines.

Reply:
I've switched to `empty` so it's more generic.